src/Client/Update/SpreadsheetUpdate.fs

```diff
@@ -39,6 +40,7 @@ module Spreadsheet =
             | _ ->
                 let newCmd =
                     if model.PersistentStorageState.Autosave then
+                        let snapshotJsonString = state.ToJsonString()
```

Comment:
`state.ToJsonString()` is one of the performance critical functions, and here it is called twice...